tsdb/engine/tsm1/engine.go

```diff
@@ -228,7 +258,8 @@ func (e *DevEngine) Begin(writable bool) (tsdb.Tx, error) {
 
 func (e *DevEngine) WriteTo(w io.Writer) (n int64, err error) { panic("not implemented") }
 
-func (e *DevEngine) writeSnapshot() error {
+// WriteSnapshot will snapshot the cache and write a new TSM file with its contents, releasing the snapshot when done.
+func (e *DevEngine) WriteSnapshot(async bool) error {
```

Comment:
>I'm not sure what having a long running goroutine dedicated to snapshots gives us. How's that different than just kicking off goroutine as needed?

From a black-box point of view, it doesn't make much difference. From an implementation point of view it is significant. Right now the goroutine to kick off snapshot-to-tsm is in the compaction goroutine. To the next guy looking at this code it's not clear whether this coupling is co-incidental, or is a key part of the design (it's the former it seems). Splitting it out into a separate goroutine -- which in turn can launch short-running goroutines if performance analysis shows us we should -- makes it crystal clear to the next guy how the components interact (or don't).

In other words it seems like unnecessary coupling of two distinct services in the engine.